src/LATBOLTZ/fix_lb_fluid.cpp

```diff
@@ -2339,59 +2350,59 @@ void FixLbFluid::SetupBuffers(void)
 //==========================================================================
 // Output fluid density and velocity to file in XDMF format
 //==========================================================================
-void FixLbFluid::dump(const int step)
+void FixLbFluid::dump(const bigint step)
 {
-  static int frameindex = 0;
+  static bigint frameindex = 0;
 
   if (dump_interval && step % dump_interval == 0) {
     // Write XDMF grid entry for time step
     if (me == 0) {
-      long int block = (long int) fluid_global_n0[0] * fluid_global_n0[1] * fluid_global_n0[2] *
-          sizeof(MPI_DOUBLE);
-      long int offset = frameindex * block * (1 + 3);
+      bigint block =
+          (bigint) fluid_global_n0[0] * fluid_global_n0[1] * fluid_global_n0[2] * sizeof(double);
```

Comment:
@CDenniston this fixes a significant bug. MPI_DOUBLE is not a data type but a constant. In OpenMPI it is (by chance) a pointer and thus 64-bit on a 64-bit machine and thus represents the same size as a double. In MPICH and its many derived MPI libraries, however, it is a 32-bit integer and thus the size is incorrect and the computed offsets are off by a factor of 2.

Also `update->ntimestep` is a `bigint`, that is a 64-bit integer when LAMMPS is compiled with default settings, but 32-bit with -DLAMMPS_SMALLSMALL. Using "long int" is not sufficient, since on some platforms, most notably Windows, it is still a 32-bit int. Since printf() formatting is messy for such cases, I've switched to using fmtlib output here which adapts to all recognized types.

Reply:
I am impressed that you caught this!  I guess this shows the value in compiling against a bunch of different systems.

Reply:
I didn't catch it. ;-)
We're running static code analysis early Friday mornings on the develop branch, that software spotted it. 
https://scan.coverity.com/

```
Please find the latest report on new defect(s) introduced to LAMMPS found with Coverity Scan.

4 new defect(s) introduced to LAMMPS found with Coverity Scan.
23 defect(s), reported by Coverity Scan earlier, were marked fixed in the recent build analyzed by Coverity Scan.

New defect(s) Reported-by: Coverity Scan
Showing 4 of 4 defect(s)


** CID 376724:    (BAD_SIZEOF)
/home/builder/workspace/dev/develop/static_analysis/cmake_coverity/lammps/src/LATBOLTZ/fix_lb_fluid.cpp: 2392 in LAMMPS_NS::FixLbFluid::dump(int)()
/home/builder/workspace/dev/develop/static_analysis/cmake_coverity/lammps/src/LATBOLTZ/fix_lb_fluid.cpp: 2379 in LAMMPS_NS::FixLbFluid::dump(int)()
/home/builder/workspace/dev/develop/static_analysis/cmake_coverity/lammps/src/LATBOLTZ/fix_lb_fluid.cpp: 2350 in LAMMPS_NS::FixLbFluid::dump(int)()


________________________________________________________________________________________________________
*** CID 376724:    (BAD_SIZEOF)
/home/builder/workspace/dev/develop/static_analysis/cmake_coverity/lammps/src/LATBOLTZ/fix_lb_fluid.cpp: 2392 in LAMMPS_NS::FixLbFluid::dump(int)()
2386                   "Dimensions=\"%d %d %d 3\">\n"
2387                   "              %s\n"
2388                   "            </DataItem>\n"
2389                   "          </DataItem>\n"
2390                   "        </Attribute>\n\n",
2391                   dx_lb / dt_lb, fluid_global_n0[2], fluid_global_n0[1], fluid_global_n0[0],
>>>     CID 376724:    (BAD_SIZEOF)
>>>     Taking the size of "&ompi_mpi_double", which is the address of an object, is suspicious.
2392                   sizeof(MPI_DOUBLE), offset + block * 1, fluid_global_n0[2], fluid_global_n0[1],
2393                   fluid_global_n0[0], dump_file_name_raw.c_str());
2394           fprintf(dump_file_handle_xdmf, "      </Grid>\n\n");
2395     
2396           frameindex++;
2397         }
```